File: src/main/java/cl/app/CategoryRules.java
package cl.app;

/**
 * Created by Daniel on 01-03-2016.
 */
public final class CategoryRules extends Rules<Category> {

    @Override
    protected void initRules() {
        rulesFor(Category.JAVA)
                .add(status -> status.getText().toLowerCase().contains("java"))
                .apply();
        rulesFor(Category.NET)
                .add(status -> status.getText().toLowerCase().contains(".net"))
                .apply();
        rulesFor(Category.JAVASCRIPT)
                .add(status -> status.getText().toLowerCase().contains("javascript"))
                .apply();
        rulesFor(Category.OTHER)
                .add(status -> {
                    String text = status.getText().toLowerCase();
                    return !(text.contains("java") || text.contains("javascript") || text.contains(".net"));
                })
                .apply();
    }

}

File: src/main/java/cl/app/rule/CategoryRules.scala
package cl.app.rule

import cl.app.`type`.Category
import cl.app.`type`.Category.{OTHER, JAVASCRIPT, NET, JAVA}
import twitter4j.Status

/**
  * Created by Daniel on 02-03-2016.
  */
object CategoryRules extends Rules[Category.Value]
  with Serializable{

  private def contain(status:Status,v:String):Boolean={
    status.getText.toLowerCase.contains(v)
  }

  override protected def initRules(): Unit = {


    forCategory(JAVA)
      .addRule(contain(_,"java"))
      .build

    forCategory(NET)
      .addRule(contain(_,".net"))
      .build

    forCategory(JAVASCRIPT)
      .addRule(contain(_,"javascript"))
      .build

    forCategory(OTHER)
      .addRule(contain(_,"php"))
      .build

  }


}
